src/server/game/Server/GameClient.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "GameClient.h"
+#include "WorldSession.h"
+#include "Unit.h"
+#include "Player.h"
+
+GameClient::GameClient(WorldSession* sessionToServer)
+{
+    _sessionToServer = sessionToServer;
+    _activelyMovedUnit = nullptr;
+}
+
+void GameClient::AddAllowedMover(Unit* unit)
+{
+    _allowedMovers.insert(unit->GetGUID());
+    unit->SetGameClientMovingMe(this);
+}
+
+void GameClient::RemoveAllowedMover(Unit* unit)
+{
+    _allowedMovers.erase(unit->GetGUID());
+    if (unit->GetGameClientMovingMe() == this)
+        unit->SetGameClientMovingMe(nullptr);
+}
+
+bool GameClient::IsAllowedToMove(Unit* unit) const
+{
+    return _allowedMovers.count(unit->GetGUID());
```

Comment:
container.find(element) != container.end() feels more appropiate

Reply:
https://en.cppreference.com/w/cpp/container/set/contains

available con c++20, not sure if we're on it

Reply:
We are on C++17